source/extensions/tracers/xray/sampling_strategy.h

```diff
@@ -15,42 +15,30 @@ namespace Tracers {
 namespace XRay {
 
 struct SamplingRequest {
-  /**
-   * Creates a new SamplingRequest
-   *
-   * @param host_name The host name the request.
-   * @param http_method The http method of the request e.g. GET, POST, etc.
-   * @param http_url The path part of the URL of the request.
-   * @param service The name of the service (user specified)
-   * @param service_type The type of the service (user specified)
-   */
-  SamplingRequest(absl::string_view host_name, absl::string_view http_method,
-                  absl::string_view http_url)
-      : host_(host_name), http_method_(http_method), http_url_(http_url) {}
-
-  const std::string host_;
-  const std::string http_method_;
-  const std::string http_url_;
+  std::string host_;
+  std::string http_method_;
+  std::string http_url_;
 };
 
 /**
  * Strategy provides an interface for implementing trace sampling strategies.
  */
 class SamplingStrategy {
 public:
-  explicit SamplingStrategy(uint64_t rng_seed) : rng_(rng_seed) {}
+  explicit SamplingStrategy(Runtime::RandomGenerator& rng) : rng_(rng) {}
```

Comment:
Is it certain that all implementations of SamplingStrategy require a random number generator? Otherwise, it makes sense to move this parameter and it's storage into the LocalizedSamplingStrategy subclass.